### ISSUE TITLE
Show products of a specific category by setting it explicitly

### ISSUE THREAD

**Issue by maartenmachiels** [OFFLINE-GmbH/oc-mall-plugin#597]
Hi Tobias,

On my homepage, I'm trying to show a slider of products "in the spotlight", a set of products that can be set manually by the admin. So I created a category named "Toppers NL", with code "toppers-nl", and tried to select products based on this code.
On my homepage layout, I've set the following configuration:

```
[products]
category = "toppers-nl"
perPage = 16
paginate = 0
includeVariants = 0
includeChildren = 0
==
```
Further on in the homepage layout, the products component is included => `{% component 'products' %}`

If I don't set the category explicitly, all products (regardless of category) are shown without issue. When setting the category with the above attribute, the page throws a 404-error. The category ID code seems to be correct though. Do you know what I am doing wrong that the products of my Toppers NL category aren't being shown on my homepage? Please note that I can visit the Toppers NL category without issue via the category pages.

Furthermore: is this the right way to get a subselection of products? I've also tried using properties, and this does work as long as the property is set as a filter. The unwanted side effect then is that this property is shown in the filters on the category page and on the detail pages. Is it possible to hide filters from category pages and detail pages, but still use them to filter out products?

Sorry if this is convoluted. I'm happy to provide additional information. Thanks in advance for your response.

**Comment by chrisvidal** [comment]
the category property should an `ID` and not a `slug`

**Comment by maartenmachiels** [comment]
Thank you Chris, so it's a numeric value, and not the code field?

**Comment by chrisvidal** [comment]
yes, it can be `2` or `13`, according to your product ID

**Comment by maartenmachiels** [comment]
This works perfectly, thanks very much!

**Comment by maartenmachiels** [comment]
Hi, I just have a small follow-up question. I'm trying to allow the client to set the category that is displayed on the homepage via a static page variable. The category information seems to show up without issue, but the products don't. The title and short description of my category are displayed, but just below, where the products should be it says: "Your selection did not return any results."

I've omitted the static page variable here for brevity and just used a fixed integer for the category to test everything out (11 is the id of an existing category). I can't seem to wrap my head around why this doesn't work. I've tried to deduct the logic from the relatedProducts component on the product detail page, but to no avail...

```
[products toppers]
perPage = 16
paginate = 0
includeVariants = 0
includeChildren = 0
==
<?php
use \OFFLINE\Mall\Models\Category as MallCategory;
function onStart() {
    $this['category'] = $category = MallCategory::with('products')->findOrFail(11);
    $this->findComponentByName('toppers')->category = $category;
}
?>
==
<h2>{{ category.name }}</h2>
{% if category.description_short | length %}
  <p>{{ category.description_short }}</p>
{% endif %}
{% component 'toppers' %}
```
